### PR TITLE
Add ServerSessionManagementService as an OSGI service

### PR DESCRIPTION
**Pull request by ThaminduR**
### Proposed changes in this pull request

Create an OSGi service for `SessionManagementService`

### CODE REVIEW

**Review by mefarazath — APPROVED** [review]

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/services/SessionManagementService.java [hunk 1]
```diff
@@ -35,19 +35,15 @@
...
+        ServerSessionManagementService serverSessionManagementService =
+                FrameworkServiceDataHolder.getInstance().getServerSessionManagementService();
+        return serverSessionManagementService.removeSession(sessionId);
     }
 
     /**
      *
      * Terminates the requested session, after validating whether the session belongs to the logged in user.
-     *
+     * Use UserSessionManagementService to terminate the session of the current user
```
Comment (piraveena) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/services/SessionManagementService.java:46]: Add deprecated java doc annotation and link to that class and method

Comment (ThaminduR) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/services/SessionManagementService.java]: Added in 26c25fc

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/ServerSessionManagementService.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+package org.wso2.carbon.identity.application.authentication.framework;
+
+/**
+ * Defines the session management service operations.
+ */
+public interface ServerSessionManagementService {
+
+    /**
+     * Terminate session related to the sessionID
```
Comment (lashinijay) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/ServerSessionManagementService.java:27]: Missing full-stops. Check other places too.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/ServerSessionManagementServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import org.wso2.carbon.identity.application.authentication.framework.ServerSessionManagementService;
+import org.wso2.carbon.identity.application.authentication.framework.cache.SessionContextCache;
+import org.wso2.carbon.identity.application.authentication.framework.context.SessionContext;
+import org.wso2.carbon.identity.application.authentication.framework.internal.FrameworkServiceDataHolder;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkUtils;
+
+/**
+ * A service to terminate the sessions of federated users
```
Comment (lashinijay) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/ServerSessionManagementServiceImpl.java:33]: Shall we end the comments with full-stops? check other places too.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/ServerSessionManagementService.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
```
Comment (sandushi) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/ServerSessionManagementService.java:4]: Shall we change `Inc` to `LLC` ? Check other places as well

**Pull request merged**